src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp

```diff
@@ -74,80 +74,109 @@ LogicalResult ONNXGenericReductionOpShapeHelper<OP_TYPE>::customComputeShape(
-      // Have already some shapes, keep them in ShapeHelper
-      DimsExpr outputDims;
-      createIE->getShapeAsDims(reduceOp.getResult(), outputDims);
+    if (isFromNone(operandAdaptor.getAxes())) {
+      // Default will be used.
+    } else if (getONNXConstantOp(operandAdaptor.getAxes())) {
+      createIE->getIntFromArrayAsSymbols(operandAdaptor.getAxes(), axes);
+    } else {
+      // When the axis is dynamic, try to infer the rank of output tensor
+      int64_t dataRank = createIE->getShapedTypeRank(operandAdaptor.getData());
+      int64_t axlesSize = createIE->getArraySize(operandAdaptor.getAxes());
+      if (!operandAdaptor.getKeepdims() && axlesSize < 0 /*undef shape*/) {
+        // Even though we did not compute the shape in ShapeHelper, return success
+        // as we gen code for dyn sizes too. Ideally, we would have the code here
+        // to compte the shape, it is currently residing in Krnl lowering.
+
+        // In fact, this case here requires further handling of unranked tensors,
+        // which I doubt we support. But there is currently a lit tests
+        // (onnx/onnx_shape_inference.mlir's test_reduce_sum_5) expect the *xf32
+        // sizes. So leave reporting success here, even though I really suspect we
+        // should report failure. return op->emitError("does not support unranked
+        // reduction output");
+        return success();
+      }
+      // With keep dim, by def the output has the same rank as the input; without
+      // keep dim, we remove 1 rank per value in the 1D axes tensor.
+      int64_t outputRank =
+          operandAdaptor.getKeepdims() ? dataRank : dataRank - axlesSize;
+      assert(outputRank >= 0 && "expected to keep at least one dim");
+
+      // Interesting idea below, namely to reuse info from output, or if not
+      // there, from input putting question mark in there. Not sure if successful,
+      // if it is, it should be generalized to all ops.
+
+      if (reduceOp.getResult().getType().template isa<RankedTensorType>()) {
+        // Have already some shapes, keep them in ShapeHelper
+        DimsExpr outputDims;
+        createIE->getShapeAsDims(reduceOp.getResult(), outputDims);
+        setOutputDims(outputDims);
+        return success();
+      }
+      // Else set is as questionmarks. Output tensor should have the same rank
+      // as the input. But size of dims is unknown.
+      DimsExpr outputDims(outputRank, QuestionmarkIndexExpr(/*isFloat*/ false));
       setOutputDims(outputDims);
       return success();
-    }
-    // Else set is as questionmarks. Output tensor should have the same rank as
-    // the input. But size of dims is unknown.
-    DimsExpr outputDims(outputRank, QuestionmarkIndexExpr(/*isFloat*/ false));
-    setOutputDims(outputDims);
-    return success();
+    } 
+  } 
+  bool noopWithEmptyAxes = false;
+  if constexpr (isAxesInput<OP_TYPE>) {
+    noopWithEmptyAxes = operandAdaptor.getNoopWithEmptyAxes() != 0;
```

Comment:
Since the high level description states that this is unsupported, should we not generate an error?